app/main/views/marketplace.py

```diff
@@ -128,7 +128,10 @@ def get_brief_by_id(framework_slug, brief_id):
     briefs = data_api_client.get_brief(brief_id)
     brief = briefs.get('briefs')
     if brief['status'] not in ['live', 'closed']:
-        abort(404, "Opportunity '{}' can not be found".format(brief_id))
+        if current_user.is_authenticated and brief['users'][0]['id'] == current_user.id and brief['status'] == 'draft':
```

Comment:
Do you need that last 'status' check - we're already inside an if statement where status has been checked